Automattic/wp-calypso
# Staging Sites: Customize Privacy section to streamline 'Public' option

Fixes https://github.com/Automattic/dotcom-forge/issues/2105
Related https://github.com/Automattic/wpcomsh/pull/1343

## Proposed Changes

Streamlines the 'Public' option for the Privacy Settings of a Staging Site:

<img width="811" alt="image" src="https://user-images.githubusercontent.com/36432/232042310-79499349-858e-41df-b533-cdbd8cb3fdf3.png">

Non-Staging Sites remain the same:

<img width="823" alt="image" src="https://user-images.githubusercontent.com/36432/230944005-430bd903-a8f7-4d83-9803-d6d0961c1215.png">

Jetpack Sites remain the same:

<img width="785" alt="image" src="https://user-images.githubusercontent.com/36432/234408021-bf719010-46fe-4193-a126-2af8daf05513.png">

## Testing Instructions

1. Create a Business site and take it Atomic.
2. Navigate to 'General Settings' and verify the 'Launch site' UI appears as expected.
3. Launch the site and verify the Privacy settings appear as expected (unchanged).
4. Navigate to 'Hosting Configuration' and create a staging site.
5. Navigate to 'General Settings' for the staging site and verify the new Privacy Settings appear as expected.
6. Spin up a jurassic.ninja site and verify the Privacy Settings appear as expected (unchanged).
7. Create a Personal site and verify the Privacy Settings appear as expected (unchanged).
